Sources/Testing/Issues/Issue.swift

```diff
@@ -179,3 +179,217 @@ extension Issue.Kind: CustomStringConvertible {
     }
   }
 }
+
+// MARK: - Codable
+
+extension Issue {
+  /// A serializable type describing a failure or warning which occurred during a test.
+  @_spi(ExperimentalSnapshotting)
+  public struct Snapshot: Sendable, Codable {
+    /// The kind of issue this value represents.
+    public var kind: Kind.Snapshot
```

Comment:
In terms of layering and design, it might be better to make `Kind.Snapshot` `Snapshot.Kind`, i.e. it's just the `Kind` of the issue snapshot. 🤔

Reply:
I don't feel strongly but then Snapshot would only be a namespace (empty enum), right? We've used the pattern I've used here for Expectation as well.

Reply:
No, `Issue.Snapshot` would still be a fully-formed nominal type, it's just we wouldn't add a separate `Kind.Snapshot`, it'd be `Kind` under `Issue.Snapshot`.

This is probably not critical to the overall design, but I want to give it a bit of thought either way.

Reply:
Oh, I misunderstood, sorry. I don't feel strongly about that either.

Reply:
Actually, I think it makes more sense to me have all "original" types have `Snapshot` types nested, instead of re-creating that same type hierarchy somewhere else.